### ISSUE TITLE
Measure's check_prerequisites is ignored when calling `$score()` on a ResampleResult

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3#1000]
Maybe a misunderstanding from my side, but shouldn't I get a warning in the code below, as the regr.debug learner does not have the `"selected_features"` property?

``` r
library(mlr3)

rr = resample(tsk("mtcars"), lrn("regr.debug"), rsmp("holdout"), store_models = TRUE)
#> INFO  [17:13:24.525] [mlr3] Applying learner 'regr.debug' on task 'mtcars' (iter 1/1)
measure = msr("selected_features")
# is default anyway
measure$check_prerequisites = "warn"

rr$score(measure)
#>    task_id learner_id resampling_id iteration selected_features
#>     <char>     <char>        <char>     <int>             <num>
#> 1:  mtcars regr.debug       holdout         1                NA
#> Hidden columns: task, learner, resampling, prediction
```

<sup>Created on 2024-02-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by be-marc** [comment]
Except if there are multiple measures that need `oob_error` or the `selected_features`.

**Comment by be-marc** [comment]
Solved #1212 #1218
